src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -4509,67 +4390,413 @@ def _get_nic_ip_config(nic, name):
+            fmt=AAZResourceIdArgFormat(
+                template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Network"
+                         "/loadBalancers/{lb_name}/backendAddressPools/{}",
+            ),
+        )
+        args_schema.lb_inbound_nat_rules = AAZListArg(
+            options=["--lb-inbound-nat-rules"],
+            arg_group="Load Balancer",
+            help="Space-separated list of names or IDs of load balancer inbound NAT rules to associate with the NIC. "
+                 "If names are used, `--lb-name` must be specified.",
+        )
+        args_schema.lb_inbound_nat_rules.Element = AAZResourceIdArg(
+            fmt=AAZResourceIdArgFormat(
+                template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Network"
+                         "/loadBalancers/{lb_name}/inboundNatRules/{}",
+            ),
+        )
+        args_schema.application_gateway_backend_address_pools._registered = False
+        args_schema.load_balancer_backend_address_pools._registered = False
+        args_schema.load_balancer_inbound_nat_rules._registered = False
+        args_schema.private_ip_allocation_method._registered = False
+        args_schema.pip_obj._registered = False
+        args_schema.asgs_obj._registered = False
+        args_schema.subnet_obj._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        args.private_ip_address_version = args.private_ip_address_version
+        args.private_ip_allocation_method = "Static" if has_value(args.private_ip_address) else "Dynamic"
+        args.application_gateway_backend_address_pools = [{"id": x} for x in args.app_gateway_address_pools] if has_value(args.app_gateway_address_pools) else None
+        args.load_balancer_backend_address_pools = [{"id": x} for x in args.lb_address_pools] if has_value(args.lb_address_pools) else None
+        args.load_balancer_inbound_nat_rules = [{"id": x} for x in args.lb_inbound_nat_rules] if has_value(args.lb_inbound_nat_rules) else None
+
+        args.pip_obj = {"id": args.public_ip_address} if has_value(args.public_ip_address) else None
+        args.asgs_obj = [{"id": x} for x in args.application_security_groups] if has_value(args.application_security_groups) else None
+
+    def pre_instance_create(self):
+        args = self.ctx.args
+        instance = self.ctx.vars.instance
+        if args.private_ip_address_version.to_serialized_data().lower() == "ipv4" and not has_value(args.subnet):
+            primary = next(x for x in instance.properties.ip_configurations if x.properties.primary)
+            args.subnet = primary.properties.subnet.id
+        if args.make_primary.to_serialized_data():
+            for config in instance.properties.ip_configurations:
+                config.properties.primary = False
+
+        args.subnet_obj = {"id": args.subnet} if has_value(args.subnet) else None
```

Comment:
Why use `args.subnet` as proxy for `args.subnet_obj`? Why not directly use `args.subnet_obj`?